lib/neo4j/core/cypher_session/adaptors/http.rb

```diff
@@ -134,22 +137,23 @@ def get(path, body = '', options = {})
 
             private
 
-            def faraday_connection
+            def faraday_connection(faraday_options = {})
+              verify_faraday_options(faraday_options,
+                                     request: [
+                                       [:basic_auth, user, password],
+                                       :multi_json
+                                     ],
+                                     response: [:multi_json, symbolize_keys: true, content_type: 'application/json']
+                                    )
               require 'faraday'
               require 'faraday_middleware/multi_json'
 
-              Faraday.new(url) do |c|
-                c.request :basic_auth, user, password
-                c.request :multi_json
-
-                c.response :multi_json, symbolize_keys: true, content_type: 'application/json'
-                c.use Faraday::Adapter::NetHttpPersistent
-
+              conn = Faraday.new(url) do |c|
                 # c.response :logger, ::Logger.new(STDOUT), bodies: true
```

Comment:
That is my understanding - all the middleware that gets called during initialization gets mounted in Faraday and these would then *always* get mounted no matter if people wanted them or not, and there would be no way to get rid of them.